File: app/src/main/java/com/example/newsapp/HeadlinesByCountry.java
package com.example.newsapp;

import android.os.Bundle;
import android.support.v4.app.FragmentManager;
import android.support.v7.app.AppCompatActivity;

import com.example.newsapp.Fragments.CountryListFragment;

public class HeadlinesByCountry extends AppCompatActivity {


    private static boolean LARGE_SCREEN_FLAG;

    private int mainFragmentContainer;
    private FragmentManager fragmentManager;


    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_country);

        if (findViewById(R.id.cl_country_detail) != null){
            this.LARGE_SCREEN_FLAG = true;
        } else{
            this.LARGE_SCREEN_FLAG = false;
        }

        loadData();
    }

    @Override
    protected void onResume() {
        super.onResume();
        if(findViewById(R.id.cl_country_detail) != null){
            this.LARGE_SCREEN_FLAG = true;

        } else {
            this.LARGE_SCREEN_FLAG = false;
        }

        loadData();
    }

    /* Loads counties to the country list fragment
    * sets up a large screen flag
    * strats up the country list fragment */

    private void loadData() {

        CountryListFragment countryListFragment = new CountryListFragment();
        Bundle bundle = new Bundle();
        bundle.putString(getString(R.string.data_flag_key),
                getString(R.string.load_country_string));

        if (LARGE_SCREEN_FLAG) {

            bundle.putBoolean(getString(R.string.large_string_flag_key)
                    , LARGE_SCREEN_FLAG);

            mainFragmentContainer = R.id.frame_main;

        } else {
            mainFragmentContainer = R.id.frame_counties_holder;
        }

        countryListFragment.setArguments(bundle);

        fragmentManager = this.getSupportFragmentManager();

        fragmentManager.beginTransaction()
                .replace(mainFragmentContainer, countryListFragment)
                .commitAllowingStateLoss();


    }
}

File: app/src/main/java/com/example/newsapp/HeadlinesByCountry.kt
package com.example.newsapp

import android.os.Bundle
import android.support.constraint.ConstraintLayout
import android.support.v4.app.FragmentManager
import android.support.v7.app.AppCompatActivity
import com.example.newsapp.Fragments.CountryListFragment

class HeadlinesByCountry : AppCompatActivity() {

    var LARGE_SCREEN_FLAG: Boolean = false
    var mainFragmentContainer: Int = 0
    lateinit var fragmentManager: FragmentManager

    override fun onCreate(savedInstanceState: Bundle?) {

        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_country)

        LARGE_SCREEN_FLAG = if (findViewById<ConstraintLayout>
                (R.id.cl_country_detail) != null) true else false
        loadData()

    }

    override fun onResume() {

        super.onResume()
        LARGE_SCREEN_FLAG = if (findViewById<ConstraintLayout>
                (R.id.cl_country_detail) != null) true else false
        loadData()
    }

    fun loadData() {

        val countryListFragment = CountryListFragment()
        val bundle = Bundle()

        bundle.putString(getString(R.string.data_flag_key), getString(R.string.load_country_string))

        if (LARGE_SCREEN_FLAG) {
            bundle.putBoolean(getString(R.string.large_string_flag_key), LARGE_SCREEN_FLAG)
            mainFragmentContainer = R.id.frame_main
        } else {
            mainFragmentContainer = R.id.frame_counties_holder
        }
        countryListFragment.arguments = bundle
        fragmentManager = this.supportFragmentManager
        fragmentManager.beginTransaction().replace(mainFragmentContainer, countryListFragment)
                .commitAllowingStateLoss()
    }

}
